cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -463,7 +463,6 @@ if(DEFINED ENV{ARROW_GBENCHMARK_URL})
 else()
   set_urls(GBENCHMARK_SOURCE_URL
            "https://github.com/google/benchmark/archive/${ARROW_GBENCHMARK_BUILD_VERSION}.tar.gz"
-           "https://github.com/ursa-labs/thirdparty/releases/download/latest/gbenchmark-${ARROW_GBENCHMARK_BUILD_VERSION}.tar.gz"
```

Comment:
I added https://github.com/ursa-labs/thirdparty/releases/download/latest/gbenchmark-v1.6.0.tar.gz to our thirdparty tarballs at https://github.com/ursa-labs/thirdparty/releases/tag/latest. Could you please retain this line? Thanks

Reply:
How about using https://apache.jfrog.io/ instead of https://github.com/ursa-labs/thirdparty/ such as https://apache.jfrog.io/artifactory/arrow/boost/1.73.0/boost_1_73_0.tar.bz2 ?

1. If GitHub is down, all mirrors aren't available.
2. https://apache.jfrog.io/ is a ASF resource but https://github.com/ursa-labs/thirdparty/ isn't a ASF resource.

We can use https://apache.jfrog.io/ with our ASF account.

Reply:
@kou 💯 agreed, we should move all the bundled third-party dependencies mirror to an ASF-managed repository such as JFrog. @bkmgit has been pushing for this recently and I think there is broad support for it. This is being tracked in ARROW-6407 and I think it would be most constructive to handle it as a consolidated effort, not do it as one-offs for each dependency.

Is it OK with you if we keep the Ursa thirdparty repo in here for now, and discuss this move to JFrog in ARROW-6407? Thanks!

Reply:
Yes.